cluster-service/pipeline.yaml

```diff
@@ -180,8 +180,50 @@ resourceGroups:
       configRef: clustersService.azureRuntimeConfig.tlsCertificatesIssuer
     - name: AVAILABILITY_ZONE_COUNT
       configRef: azureRegionAvailabilityZoneCount
-    - name: DEPLOY_DEBUG_JOB
-      configRef: clustersService.deployDebugJob
+    shellIdentity:
+      input:
+        resourceGroup: global
+        step: output
+        name: globalMSIId
+  - name: deploy-cs-debug-jobs
+    aksCluster: '{{ .svc.aks.name }}'
+    action: Shell
+    command: make deploy-cs-debug-jobs
+    workingDir: ./
+    dryRun:
+      variables:
+      - name: DRY_RUN
+        value: "true"
+    dependsOn:
+    - resourceGroup: global
+      step: mirror-image
+    - resourceGroup: service
```

Comment:
I understand this means that the cs-debug-jobs pipeline step will be executed only and only if the deploy step is fully completed **and** is completed successfully. Is this understanding correct?

I am trying to make sure that we have the guarantee that the K8s Job(s) will only be executed in that scenario.

Reply:
yes